From cf6b6080951e9551134d91ab992d7de068b9d553 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Mon, 4 Jun 2018 10:05:28 +0200
Subject: GCC: do not initialize on .h template url ScopedObserver

After a recent change, OmniboxViewViews initializes a ScopedObserver
of TemplateURLService in .h, using a forward declaration. This is
not valid as ScopedObserver needs full declaration for its destructor.

So move initialization to .cc. This fixes GCC build.

Change-Id: I1e5185319980bb558c24546b0e6d104990b88dfa

diff --git a/chrome/browser/ui/views/omnibox/omnibox_view_views.cc b/chrome/browser/ui/views/omnibox/omnibox_view_views.cc
index f541923a192a..65b7e91ccd42 100644
--- a/chrome/browser/ui/views/omnibox/omnibox_view_views.cc
+++ b/chrome/browser/ui/views/omnibox/omnibox_view_views.cc
@@ -134,7 +134,8 @@ OmniboxViewViews::OmniboxViewViews(OmniboxEditController* controller,
       select_all_on_mouse_release_(false),
       select_all_on_gesture_tap_(false),
       latency_histogram_state_(NOT_ACTIVE),
-      friendly_suggestion_text_prefix_length_(0) {
+      friendly_suggestion_text_prefix_length_(0),
+      scoped_template_url_service_observer_(this) {
   set_id(VIEW_ID_OMNIBOX);
   SetFontList(font_list);
 
diff --git a/chrome/browser/ui/views/omnibox/omnibox_view_views.h b/chrome/browser/ui/views/omnibox/omnibox_view_views.h
index 5d151e98e99d..62ec262172a2 100644
--- a/chrome/browser/ui/views/omnibox/omnibox_view_views.h
+++ b/chrome/browser/ui/views/omnibox/omnibox_view_views.h
@@ -336,7 +336,7 @@ class OmniboxViewViews : public OmniboxView,
   ScopedObserver<ui::Compositor, ui::CompositorObserver>
       scoped_compositor_observer_{this};
   ScopedObserver<TemplateURLService, TemplateURLServiceObserver>
-      scoped_template_url_service_observer_{this};
+      scoped_template_url_service_observer_;
 
   DISALLOW_COPY_AND_ASSIGN(OmniboxViewViews);
 };
-- 
2.14.1

